pkg/api/controller.go

```diff
@@ -3941,6 +3938,15 @@ func (c *Controller) MergeIntoBranch(w http.ResponseWriter, r *http.Request, bod
 	})
 }
 
+func getCommitter(user *model.User) string {
```

Comment:
1. Can we make it a method on the model
2. We should have email verification before user's creation - we should assume that email is valid. You can keep the null and empty check before return the value.
3. Need to verify ldap integration as which value we keep for username, not sure email is populated in this case and maybe we need too apply the friendly name in this case. Alternative, open a specific issue for external auth with ldap.

Reply:
1. changing 
2. I know, but still we are getting the email from an external service, thought it's better to stay safe than have unknown strings in the committer, @nopcoder WDYT?
3. Even if that is the case, will we prefer the friendly name that is editable to the username?

Reply:
My concern is that if the case the user's email is not an actual email, have a problem in some place else and we don't want to move this problem (and verify it) every time we use information from user model.

Prefer the username.

Reply:
Thanks! Changed PTAL